test/integration/smoke/test_hypervisor_capabilities.py

```diff
@@ -0,0 +1,291 @@
+        except Exception as e:
+            self.debug("Failed to attach 2nd DATA disk when max DATA disk limit was set to 1: %s" % e)
+
+        self.updateHostHypervisorCapability(self.hostCapability.id, 32)
+
+        vm.attach_volume(
+            self.userapiclient,
+            volume_created2
+        )
+        vm.detach_volume(self.userapiclient, volume_created1)
+        vm.detach_volume(self.userapiclient, volume_created2)
+
+    @skipTestIf("notSupported")
+    @attr(tags=["advanced", "basic"], required_hardware="false")
+    def test_01_check_hypervisor_vm_snapshot(self):
+        """ Test hypervisor vmsnapshotenabled effect
+
+        # 1. Set vmsnapshotenabled to false for hypervisor
+        # 2. List capabilities and verify value
+        # 3. Deploy a VM
+        # 4. Try VM snapshot, it should fail
+        # 5. Set vmsnapshotenabled to true for hypervisor
+        # 6. Try VM snapshot again, it should succeed
+        """
+        self.updateHostHypervisorCapability(self.hostCapability.id, None, False)
+        capabilities = self.listHostHypervisorCapabilities(self.hostCapability.id)
+        self.assertTrue(isinstance(capabilities, list), "listHypervisorCapabilities response not a valid list")
+        self.assertEqual(len(capabilities), 1, "listHypervisorCapabilities response not valid")
+        self.assertEqual(capabilities[0].vmsnapshotenabled,
+            False,
+            "listHypervisorCapabilities response vmsnapshotenabled value not False")
+
+        Host.update(self.apiclient, id=self.host.id, hosttags="host1")
+        vm = VirtualMachine.create(
+            self.userapiclient,
+            self.services["small"],
+            templateid=self.template.id,
+            accountid=self.account.name,
+            domainid=self.account.domainid,
+            serviceofferingid=self.service_offering.id,
+            zoneid=self.zone.id
+        )
+
+        try:
+            fail_snapshot = VmSnapshot.create(
+                self.userapiclient,
+                vmid=vm.id,
+                snapshotmemory="false",
+                name="Test Snapshot",
+                description="Test Snapshot Desc"
+            )
+            if vm_snapshot:
+                VmSnapshot.deleteVMSnapshot(self.userapiclient, fail_snapshot.id)
+            self.fail("Successful to take VM snapshot even when vmsnapshotenabled was set to False")
+        except Exception as e:
+            self.debug("Failed to take VM snapshot even vmsnapshotenabled was set to False: %s" % e)
+
+        self.updateHostHypervisorCapability(self.hostCapability.id, None, True)
+
+        vm_snapshot = VmSnapshot.create(
+            self.userapiclient,
+            vmid=vm.id,
+            snapshotmemory="false",
+            name="Test Snapshot",
+            description="Test Snapshot Desc"
+        )
+        if vm_snapshot:
+            VmSnapshot.deleteVMSnapshot(self.userapiclient, vm_snapshot.id)
```

Comment:
```suggestion
        self.cleanup.append(vm_snapshot)
```

Reply:
@DaanHoogland this doesn't work

Reply:
had to make change to `lib/base.py`

Reply:
ok, hadn;t realised that, but good solution :+1: